controller/SynchronisationApi.php

```diff
@@ -94,12 +102,22 @@ public function fetchEntityDetails()
             $type = $parameters[self::PARAM_TYPE];
             $entityIds = $parameters[self::PARAM_ENTITY_IDS];
             $entityIds = is_array($entityIds) ? $entityIds : [$entityIds];
-            $params = isset($parameters[self::PARAM_PARAMETERS]) ? $parameters[self::PARAM_PARAMETERS] : [];
+            if (isset($parameters[self::PARAM_PARAMETERS])) {
+                $params = $parameters[self::PARAM_PARAMETERS];
+            }
+
+            $eventManager->trigger(new SyncRequestEvent($params, $report));
 
-            $this->returnJson($this->getSyncService()->fetchEntityDetails($type, $entityIds, $params));
+            $entities = $this->getSyncService()->fetchEntityDetails($type, $entityIds, $params);
+            $report->setData($logData = [$type => ['pushed' => count($entities)]]);
+            $report->add(\common_report_Report::createInfo(sprintf('(%s) %d entities pushed.', $type, count($entities))));
 
+            $this->returnJson($entities);
         } catch (\Exception $e) {
+            $report->add(\common_report_Report::createFailure('Synchronization request failed: ' . $e->getMessage()));
             $this->returnFailure($e);
+        } finally {
+            $eventManager->trigger(new SyncResponseEvent($params, $report));
```

Comment:
this point will never be reached if an exception is thrown, because of `exit` inside of `$this->returnFailure($e);`